packages/react-centra-checkout/src/Context/index.tsx

```diff
@@ -0,0 +1,317 @@
+import * as React from 'react'
+import * as Centra from '@noaignite/centra-types'
+import ApiClient from '../ApiClient'
+
+const apiClient = ApiClient.default
+
+/** The prop types that CentraProvider accepts */
+export interface ProviderProps {
+  apiUrl: string
+  children: React.ReactNode
+  disableInitialSelection: boolean
+  paymentFailedPage: string
+  paymentReturnPage: string
+}
+
+export interface ContextMethods {
+  addItem?(item: string, quantity?: number): Promise<Response>
+  addNewsletterSubscription?(email: string): Promise<Response>
+  addVoucher?(voucher: string): Promise<Response>
+  decreaseCartItem?(line: string): Promise<Response>
+  increaseCartItem?(line: string): Promise<Response>
+  init?(selectionData?: Centra.SelectionResponse): Promise<void>
+  removeCartItem?(line: string): Promise<Response>
+  removeVoucher?(voucher: string): Promise<Response>
+  submitPayment?(data: Record<string, unknown>, locale: string): Promise<Record<string, unknown>>
+  updateCountry?(country: string, data: { language: string }): Promise<Response>
+  updateCartItemQuantity?(line: string, quantity: number): Promise<Response>
+  updateCartItemSize?(cartItem: Centra.SelectionItemModel, item: string): Promise<Response>
+  updateLanguage?(language: string): Promise<Response>
+  updatePaymentMethod?(paymentMethod: string): Promise<Response>
+  updateShippingMethod?(shippingMethod: string): Promise<Response>
+}
+
+export interface ContextProperties extends ContextMethods, Centra.SelectionResponse {
+  apiUrl?: string
+}
+
+const DEFAULT_VALUE = {
+  countries: [],
+  languages: [],
+  location: {},
+  paymentFields: {
+    address: {},
+    shippingAddress: {},
+    termsAndConditions: {},
+  },
+  paymentMethods: [],
+  selection: {
+    address: {},
+    discounts: {},
+    items: [],
+    totals: {},
+  },
+  shippingMethods: [],
+}
```

Comment:
I think these should have a more descriptive name like `SELECTION_INITIAL_VALUE` or so and also exported for certain occasions @alexanderflink